src/Transports.AspNetCore/WebSockets/WebSocketHandler.cs

```diff
@@ -0,0 +1,165 @@
+        IHostApplicationLifetime hostApplicationLifetime,
+        IWebSocketAuthenticationService? authorizationService = null)
+    {
+        _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
+        _executer = executer ?? throw new ArgumentNullException(nameof(executer));
+        _serviceScopeFactory = serviceScopeFactory ?? throw new ArgumentNullException(nameof(serviceScopeFactory));
+        Options = options ?? throw new ArgumentNullException(nameof(options));
+        AuthorizationOptions = authorizationOptions ?? throw new ArgumentNullException(nameof(authorizationOptions));
+        _hostApplicationLifetime = hostApplicationLifetime ?? throw new ArgumentNullException(nameof(hostApplicationLifetime));
+        _authorizationService = authorizationService;
+    }
+
+    /// <inheritdoc/>
+    public virtual async Task ExecuteAsync(HttpContext httpContext, WebSocket webSocket, string subProtocol, IUserContextBuilder userContextBuilder)
+    {
+        if (httpContext == null)
+            throw new ArgumentNullException(nameof(httpContext));
+        if (webSocket == null)
+            throw new ArgumentNullException(nameof(webSocket));
+        if (subProtocol == null)
+            throw new ArgumentNullException(nameof(subProtocol));
+        if (userContextBuilder == null)
+            throw new ArgumentNullException(nameof(userContextBuilder));
+        var appStoppingToken = _hostApplicationLifetime.ApplicationStopping;
+        using var cts = CancellationTokenSource.CreateLinkedTokenSource(httpContext.RequestAborted, appStoppingToken);
+        if (cts.Token.IsCancellationRequested)
+            return;
+        try
+        {
+            using var webSocketConnection = CreateWebSocketConnection(httpContext, webSocket, cts.Token);
+            using var operationMessageReceiveStream = CreateReceiveStream(webSocketConnection, subProtocol, userContextBuilder);
+            await webSocketConnection.ExecuteAsync(operationMessageReceiveStream);
+        }
+        catch (OperationCanceledException) when (appStoppingToken.IsCancellationRequested)
+        {
+            // terminate all pending WebSockets connections when the application is in the process of stopping
+
+            // note: we are consuming OCE in this case because ASP.NET Core does not consider the task as canceled when
+            // an OCE occurs that is not due to httpContext.RequestAborted; so to prevent ASP.NET Core from considering
+            // this a "regular" exception, we consume it here
+        }
+    }
+
+    /// <summary>
+    /// Creates an <see cref="IWebSocketConnection"/>, a WebSocket message pump.
+    /// </summary>
+    protected virtual IWebSocketConnection CreateWebSocketConnection(HttpContext httpContext, WebSocket webSocket, CancellationToken cancellationToken)
+        => new WebSocketConnection(httpContext, webSocket, _serializer, Options, cancellationToken);
```

Comment:
`_serializer` is not available for overrides without resaving it into field. Let's add `protected` property.

Reply:
The same concern about overrides for `CreateReceiveStream` but of course one may just resave all needed arguments into fields.

Reply:
Ok, I can make protected properties for all fields... (??)